Allow asynchronously cleanup journal (#857)
toptal/chewy: README.md

rake chewy:journal:apply["$(date -v-1H -u +%FT%TZ)",users] # apply journaled changes for the past hour on UsersIndex only
```

> For the cases when journal has grown up to enormous size, classical way of deletion is not quite possible. Fortunately chewy internally uses [delete-by-query](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/docs-delete-by-query.html#docs-delete-by-query-task-api) ES function which support async execution with batching and [throttling](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-delete-by-query.html#docs-delete-by-query-throttle).

Reviewer: ```suggestion
When the size of the journal becomes very large, the classical way of deletion is not possible. Fortunately, Chewy internally uses [delete-by-query](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/docs-delete-by-query.html#docs-delete-by-query-task-api) ES function which supports async execution with batching and [throttling](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-delete-by-query.html#docs-delete-by-query-throttle).
```

> For the cases when journal has grown up to enormous size, classical way of deletion is not quite possible. Fortunately chewy internally uses [delete-by-query](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/docs-delete-by-query.html#docs-delete-by-query-task-api) ES function which support async execution with batching and [throttling](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-delete-by-query.html#docs-delete-by-query-throttle).

Reviewer 2: _Impossible is nothing_

it would make "classical way of deleting" obstructive and time consuming.

> Available options are listed below, they can by set by ENV variables:

Reviewer: ```suggestion
The available options, which can be set by ENV variables, are listed below:
```

> * `WAIT_FOR_COMPLETION` - boolean flag, options controls async execution, It waits by default, when set to `false`, Elasticsearch performs some preflight checks, launches the request, and returns a task you can use to cancel or get the status of the task

Reviewer: ```suggestion
* `WAIT_FOR_COMPLETION` - a boolean flag. It controls async execution. It waits by default. When set to `false`, Elasticsearch performs some preflight checks, launches the request, and returns a task you can use to cancel the task or get its status.
```

Reviewer 3: > returns a task you can use to cancel the task or get its status.

maybe

> returns a special task you can use to cancel the deletion task or get the status of the deletion task.

I find a bit confusing to have two different tasks referred in the same sentence.

Reviewer 3: Also, in the code we use `FALSE_VALUES = %w[0 f false off].freeze`, could we document here, how to set a `false` value?

Reviewer 2: This is the same task. In the response you are getting task (reference - an id) that you can use to can use when requesting the task in Task API.

> * `REQUESTS_PER_SECOND` - float, The throttle for this request in sub-requests per second

Reviewer: ```suggestion
* `REQUESTS_PER_SECOND` - float. The throttle for this request in sub-requests per second.
```

> * `SCROLL_SIZE` - integer, Size of the scroll request that powers the operation

Reviewer: ```suggestion
* `SCROLL_SIZE` - integer. The size of the scroll request that powers the operation.
```

Reviewer: I'm not quite sure what that means. :)

Reviewer 3: I think that we could 2-3 sentences of explaination of how the delete by query works in https://github.com/toptal/chewy/pull/857/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R1149. I'd describe what it is and how is uses the bulk API. With such an introduction it might be easier to understand all the params that we expose as ENVs. WDYT?

Reviewer 2: The way `delete-by-query` works in described in the link. If you don't understand the options of `delte-by-query` because you don't know how it works you reach ES documentation. I don't see much point of copying ES documentation here.

> * `SCROLL_SIZE` - integer, Size of the scroll request that powers the operation

Reviewer 3: I'd also document explicitly what are the default values if we don't set the config ENV vars.

```bash

> rake chewy:journal:clean WAIT_FOR_COMPLETION=false REQUESTS_PER_SECOND=10 SCROLL_SIZE=5000

Reviewer 4: If the `REQUESTS_PER_SECOND` is a float, maybe you should have an example with a float value not an integer.

Reviewer 2: This parameter is floating point number as it is the type accepted by the API. It don't see much improvement in presenting some value that would be used very unlikely. This also shows that you can pass the value as plain integer and it will be converted to float for API call.
